### PR TITLE
Support for Local Encryption/Decryption

### PR DESCRIPTION
**Pull request by m4dc4p**
* Exposed `get_access_key`, `put_access_key` and
  `delete_access_key` methods.
* Added `encrypt_record`, `encrypt_new_record` and `decrypt_record`
  methods to module `Crypto`, allowing local encryption & decryption.
* Refactoring to align with Ruby idioms.

### CODE REVIEW

**Comment by m4dc4p** [comment]
For https://toznysecurity.atlassian.net/browse/E3DB-795.

File: lib/e3db/client.rb [hunk 1]
```diff
@@ -449,18 +537,26 @@ def each
...
-            record = Record.new(meta: r[:meta], data: r[:record_data] || Hash.new)
-            if q.include_data && !@raw
-              access_key = r[:access_key]
-              if access_key
-                record = @client.instance_eval {
-                  ak = decrypt_eak(access_key)
-                  decrypt_record_with_key(record, ak)
-                }
+            if q.include_data
+              if !@raw
```
Comment (ericmann) [inline lib/e3db/client.rb]: Thought we were getting rid of the `raw` flag in queries?

Comment (m4dc4p) [reply lib/e3db/client.rb]: Will happen in a follow up PR (see https://github.com/tozny/e3db-ruby/tree/jgbailey/safe-delete  branch).

Comment (ericmann) [reply lib/e3db/client.rb]: :+1:

**Comment by m4dc4p** [comment]
Re-opening as this got re-based and re-worked.

**Pull request closed without merging**